tests_e2e/playwright/conftest.py

```diff
@@ -0,0 +1,46 @@
+import contextlib
+import logging
+import os
+
+import dotenv
+import pytest
+from navigator import Navigator
+
+logger = logging.getLogger()
+
+
+@pytest.fixture(scope="session")
+def _navigator_session(browser_name):
+    """Set up a navigator instance, login with username/password, start
+    a server. Teardown when session is complete.
+    Do not use this for individual tests, use `navigator` fixture
+    for tests."""
+    dotenv.load_dotenv()
+    # try/except added here in attempt to reach teardown after error in
+    # order to close the browser context which will save the video so I debug
+    # the error.
+    try:
+        nav = Navigator(
+            nebari_url=os.environ["NEBARI_FULL_URL"],
+            username=os.environ["USERNAME"],
+            password=os.environ["PASSWORD"],
+            headless=False,
+            browser=browser_name,
+            auth="password",
+            instance_name="small-instance",  # small-instance included by default
+            video_dir="videos/",
+        )
+        nav.login_password()
+        nav.start_server()
+        yield nav
+    except Exception as e:
+        logger.debug(e)
+        with contextlib.suppress(NameError):
+            nav.teardown()
```

Comment:
IIUC, creating a `Navigator` instance just spawns a browser. Meaning, the only thing that can go wrong there is a wrong browser name (error on our side) or the browser cannot be spawned (error on `playwright` side). If that is correct, I would prefer two `try / except` blocks.

Functionally that won't make a difference, but I think it makes the implementation easier to understand. While correct,

```python
with contextlib.suppress(NameError):
    nav.teardown()
```
is a little too much for my taste.

```suggestion
        try:
        nav = Navigator(
            nebari_url=os.environ["NEBARI_FULL_URL"],
            username=os.environ["USERNAME"],
            password=os.environ["PASSWORD"],
            headless=False,
            browser=browser_name,
            auth="password",
            instance_name="small-instance",  # small-instance included by default
            video_dir="videos/",
        )
    except Exception as e:
        logger.debug(e)
        raise

    try:
        nav.login_password()
        nav.start_server()
        yield nav
    except Exception as e:
        logger.debug(e)
        raise
    finally:
        nav.teardown()
```